product_filters/models/product.py

```diff
@@ -74,54 +74,56 @@ def _get_async_ranges(self, category):
-        if category:
-            prod_ids = prod_obj.search(
-                [('public_categ_ids', 'child_of', int(category)),
-                 ('website_published', '=', True)])
-        if prod_ids:
-            for prod in prod_ids:
-                for ran in ranges_obj:
-                    if ran.upper > prod.list_price > ran.lower:
-                        if ran.id in count_dict.keys():
-                            count_dict[ran.id] += 1
-                        else:
-                            count_dict[ran.id] = 1
-                    if ran.id not in count_dict.keys():
-                        count_dict[ran.id] = 0
-            to_jsonfy = [{'id': k, 'qty': count_dict[k]} for k in count_dict]
-            return to_jsonfy
+        if not category:
+            return []
+        prod_ids = prod_obj.search(
+            [('public_categ_ids', 'child_of', int(category)),
+             ('website_published', '=', True)])
+        if not prod_ids:
+            return []
+        for prod in prod_ids:
+            for ran in ranges_obj:
+                if ran.upper > prod.list_price > ran.lower:
+                    if ran.id in count_dict.keys():
+                        count_dict[ran.id] += 1
+                    else:
+                        count_dict[ran.id] = 1
+                if ran.id not in count_dict.keys():
+                    count_dict[ran.id] = 0
+        to_jsonfy = [{'id': k, 'qty': count_dict[k]} for k in count_dict]
+        return to_jsonfy
 
     @api.model
     def _get_async_values(self, category):
         prod_obj = self.env['product.template']
         count_dict = {}
         prod_ids = []
-        if category:
-            prod_ids = prod_obj.search(
-                [('public_categ_ids', 'child_of', int(category)),
-                 ('website_published', '=', True)])
-        if prod_ids:
-            for prod in prod_ids:
-                for line in prod.attribute_line_ids:
-                    for value in line.value_ids:
-                        if value.id in count_dict.keys():
-                            count_dict[value.id] += 1
-                        else:
-                            count_dict[value.id] = 1
-                        if value.id not in count_dict.keys():
-                            count_dict[value.id] = 0
-            to_jsonfy = [{'id': k, 'qty': count_dict[k]} for k in count_dict]
-            return to_jsonfy
+        if not category:
+            return []
+        prod_ids = prod_obj.search(
+            [('public_categ_ids', 'child_of', int(category)),
+             ('website_published', '=', True)])
+        if not prod_ids:
+            return []
+        for prod in prod_ids:
+            for line in prod.attribute_line_ids:
+                for value in line.value_ids:
+                    if value.id in count_dict.keys():
+                        count_dict[value.id] += 1
+                    else:
+                        count_dict[value.id] = 1
```

Comment:
I don't understand very well the logic here but...
What about use `count_dict.set_default(value.id, 0);count_dict[value.id] += 0`?
<img width="689" alt="screen shot 2016-07-08 at 12 45 54 pm" src="https://cloud.githubusercontent.com/assets/6644187/16696323/013afea2-450a-11e6-983e-981b9e03afed.png">